lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -38,7 +38,7 @@
     }
 
     while (k < len) {
-        if (o[k] === searchElement && (o[k] !== undefined || searchElement !== undefined)) {
+        if (o[k] === searchElement && (o[k] !== undefined || (searchElement === undefined && o.hasOwnProperty(k)))) {
```

Comment:
^ oh that's tricky. I was going to say that for arrays, holes in the set of own properties are just undefined and the prototype is not checked. But then I remembered that that weirdness is only true if the indexes are outside the length of the array or are non-integer numbers (like 3.14). I think. See https://github.com/Microsoft/ChakraCore/pull/2640 and the linked issue it resolves.